serverless/containers/how-to/add-a-custom-domain-to-a-container.mdx

```diff
@@ -36,11 +36,65 @@ Each container comes with an automatically generated domain, on which it is avai
+2. Before continuing, make sure the DNS record is propagated. If not, the custom domain will likely end up in `error` on step 6. You can check this by:
+    - running `dig`, `drill` or `nslookup` commands on your domain using common DNS resolvers `1.1.1.1` and `8.8.8.8`: `dig <your_custom_domain> @1.1.1.1`, `dig <your_custom_domain> @8.8.8.8`
+    - or by using the DNS check propagation tool of your choice (some are available online)
+3. Click the **Endpoints** tab on your containers' page.
+4. Click **+ Add endpoint**.
+5. Enter your custom domain name and validate the form.
+6. The console runs checks on the CNAME and TLS configuration of the custom domain name. See technical details below for more details.
+7. Once the custom domain status is `ready`, you can connect to your container through that custom domain.
 
 <Message type="important">
   **Custom Domains** do not currently support websockets. If you need to use websockets, you must configure them with the original container URL.
 </Message>
+
+## Technical details and troubleshooting
+
+Read this section if you want more details about how custom domains are handled or if you are trying to troubleshoot an issue.
+
+When adding a custom domain, the following tasks will be performed on Scaleway's side:
+
+1. Configure the custom domain on our gateways, so they know that they should handle traffic for that custom domain.
+2. Ensure that the custom domain points to one of the Serverless Container existing endpoints: either the default one `....functions.fnc.<region>.scw.cloud`, or an already existing custom domain.
+3. If step 2 is OK, generate a dedicated TLS certificate for that custom domain so it can answer to HTTPS requests. Note: **during** that step, the custom domain might be available in plain HTTP, but it will be accessible in HTTPS only once this step is over.
+
+During the configuration of a custom domain (`pending` status), callers will likely receive HTTP 404 errors, while our gateways are reconfiguring themselves.
+
+The last 2 steps above are done simultaneously with a HTTP-01 challenge. If that challenge does not succeed after 3 minutes:
+
+- The custom domain will be put in `error` status.
+- The custom domain will not be reachable (in HTTP or HTTPS).
+- No TLS certificate will be generated.
+
+HTTP-01 challenge failure (and by extension, a custom domain in `error` status) can happen for the following reasons (outside of internal errors):
+
+| Issue | Description | How to fix or avoid this? |
+| ----- | ----------- | ----------- |
+| DNS record is not correctly configured. | If the DNS record (mostly CNAME) is misconfigured on your DNS provider side, and the custom domain to the Serverless Container default endpoint, we will not be able to configure the custom domain. | Ensure that on your DNS provider, a CNAME record links your custom domain to the Serverless Container. |
```

Comment:
note: Dumb remark, but I feel like the `How to` is a rewording of the `Issue` column. Could we link to the doc/pic on what the CNAME record should look like?

Reply:
Yep, I'm not sure pictures are great in the table because they are too small, so I'll just add a text:

```suggestion
| DNS record is not correctly configured. | If the DNS record (mostly CNAME) is misconfigured on your DNS provider side, and the custom domain to the Serverless Container default endpoint, we will not be able to configure the custom domain. | Ensure that on your DNS provider, a CNAME record links your custom domain to the Serverless Container. To test, you can run the following command: `dig <your_custom_domain>` (or `nslookup`), and make sure that the value returned is a `CNAME` record to `<your_function>.functions.fnc.<region>.scw.cloud`. |
```